tabled_derive/src/lib.rs

```diff
@@ -618,9 +683,26 @@ fn merge_attributes(attr: &mut Attributes, global_attr: &StructAttributes) {
     }
 }
 
-fn fnarg_tokens(arg: &FuncArg) -> TokenStream {
+fn fnarg_tokens(
+    arg: &FuncArg,
+    fields: &Fields,
+    field_name: impl Fn(usize, &Field) -> TokenStream,
+) -> TokenStream {
     match arg {
         FuncArg::SelfRef => quote! { &self },
+        FuncArg::SelfProperty(val) => {
+            let property_name = syn::Ident::new(val, proc_macro2::Span::call_site());
+
+            // We find the field in the neighbours instead of taking self, which would be the top object.
+            // This is for nested formatting.
+            for (i, field) in fields.iter().enumerate() {
+                let field_name_result = field_name(i, field);
+                if field_name_result.to_string() == val.to_string() {
+                    return quote! { #field_name_result };
+                }
+            }
```

Comment:
Why exactly we need it?

(SRE, just cant crack it fastly)

Reply:
Passing the fields like this, we can associate the right properties. Otherwise, self could represent the highest level struct if we are trying to format the property of a nested struct.

Is it more clear? The test "format" on enums is verifying this issue.